app/lib/dfe_data_tables/categories_parser.rb

```diff
@@ -42,17 +42,21 @@ def categories
 
         concept_hash = concept(row)
 
-        if row[3].present?
+        l0 = obj&.last
+        l1 = l0&.dig(:subcat)&.last
+        l2 = l1&.dig(:subcat)&.last
+        l3 = l2&.dig(:subcat)&.last
+        unless l3.nil?
           obj&.last&.dig(:subcat)&.last&.dig(:subcat)&.last&.dig(:subcat)&.last&.dig(:concepts)&.push(concept_hash)
```

Comment:
Could this just be `l3.dig(:concepts)&.push(concept_hash)`?

Reply:
Unless I missed something with Ruby and instantiating variables I don't think doing that that will update `obj`, which is the aggregator item that we need to update because it gets carried through and returned by the `each_with_object` loop.

Reply:
I think `l3` becomes a reference to that point in the hash, so it should update it. As a test:

```ruby
obj = [
	{
		subcat: [
			{
				subcat: [
					{
						subcat: [
							{
								concepts: [
									:test
								]
							}
						]
					}
				]
			}
		]
	}
]
l0 = obj&.last
l1 = l0&.dig(:subcat)&.last
l2 = l1&.dig(:subcat)&.last
l3 = l2&.dig(:subcat)&.last

obj&.last&.dig(:subcat)&.last&.dig(:subcat)&.last&.dig(:subcat)&.last&.dig(:concepts)&.push(:from_obj)

l3.dig(:concepts)&.push(:from_l3)

puts "l3:"
puts l3.dig(:concepts)

puts "" 

puts "obj"
puts obj&.last&.dig(:subcat)&.last&.dig(:subcat)&.last&.dig(:subcat)&.last&.dig(:concepts)
```

outputs:

```
l3: 
[:test, :from_obj, :from_l3]

obj:
[:test, :from_obj, :from_l3]
```

Reply:
It looks like `l3` becomes a reference to the object within the `obj` hash, but we can test that:

```ruby
obj = [
	{
		subcat: [
			{
				subcat: [
					{
						subcat: [
							{
								concepts: [
									:test
								]
							}
						]
					}
				]
			}
		]
	}
]
l0 = obj&.last
l1 = l0&.dig(:subcat)&.last
l2 = l1&.dig(:subcat)&.last
l3 = l2&.dig(:subcat)&.last

obj&.last&.dig(:subcat)&.last&.dig(:subcat)&.last&.dig(:subcat)&.last&.dig(:concepts)&.push(:from_obj)

l3.dig(:concepts)&.push(:from_l3)

puts "l3:"
puts l3.dig(:concepts)

puts "" 

puts "obj"
puts obj&.last&.dig(:subcat)&.last&.dig(:subcat)&.last&.dig(:subcat)&.last&.dig(:concepts)
```

outputs:

```
l3: 
[:test, :from_obj, :from_l3]

obj:
[:test, :from_obj, :from_l3]
```

Reply:
You are right, changed.